posthog/middleware.py

```diff
@@ -145,11 +145,31 @@ def __init__(self, get_response):
 
     def __call__(self, request: HttpRequest):
         if request.user.is_authenticated:
+            path_parts = request.path.strip("/").split("/")
+            project_id_in_url = None
+            if len(path_parts) >= 2 and path_parts[0] == "project" and path_parts[1].isdigit():
+                project_id_in_url = int(path_parts[1])
+            elif (
+                len(path_parts) >= 3
+                and path_parts[0] == "api"
+                and path_parts[1] == "project"
+                and path_parts[2].isdigit()
+            ):
+                project_id_in_url = int(path_parts[2])
+
+            if (
+                project_id_in_url is not None
+                and request.user.team is not None
+                and request.user.team.pk != project_id_in_url
+            ):
+                new_team = request.user.teams.get(pk=project_id_in_url)
+                self.switch_team(new_team, request)
```

Comment:
This is kind of confusing because now we have `switch_team` as well as `switch_team_if_needed_and_possible`, when in this branch we also switch team if needed and possible _without_ `switch_team_if_needed_and_possible`